src/microbe_stage/Microbe.cs

```diff
@@ -1373,6 +1373,35 @@ public void UnbindAll()
         Colony?.RemoveFromColony(this);
     }
 
+    /// <summary>
+    /// Recalculates exterior organelles according to the true middle of the cell
+    /// </summary>
+    public void RecalculateExteriorOrganellesPosition()
+    {
+        foreach (var organelle in organelles.Organelles)
+        {
+            if ((organelle.Definition.Name == "Flagellum" || organelle.Definition.Name == "Pilus") &&
```

Comment:
this is the troublemaker with languages

Reply:
LOOOOOOOOOL

Reply:
Is there a variable that hold the name for those ?Like aconstant?

Reply:
use `InternalName`

Reply:
or even better:
`organelle.HasComponentFactory<ExternallyPositionedComponent>()`

Reply:
Definitely should use the actual info on what this *should* run on.
If exact names should be checked, which probably isn't needed here, but I'll comment for the future: use `var rusticyanin = simulation.GetOrganelleType("rusticyanin");` and then compare for example `if(organelle == rusticyanin)`

Reply:
Is this some kind of special loop to process external organelles? Why not use the normal organelle Update callback for whatever these need to do?

Reply:
I need it to get the other organelles position.SO that even if you place the external components inside the cell they will be placed outside the cell at editor exit/preview

Reply:
Why is that being done?
It's been this way forever:
- External organelles always place their visual model at the membrane edge, no matter if they are an edge organelle or not.

Reply:
Their orientation would fail,if i didnt.

Reply:
How, why? Why not just fix the external organelle visual positioning math, like I suggested?

Reply:
The math in the `Update` doesnt have access to others organelles position,and i would have to modify the` update `function of the organelles to take the middle of the cell into account,which i thought wouldnt be that nice.

Reply:
The middle of the cell is always 0,0. at least once https://github.com/Revolutionary-Games/Thrive/issues/1645 is complete.
I would much prefer to have the calculation be:
- just normal when the flagella is not at 0,0
- If the flagella ends up being at 0,0 *then* special code kicks in that looks to see if the flagellum should point backwards or forwards determined by looking at the Microbe the organelle is attached to
Would that approach have a problem?

Reply:
Its a problem if the flagella is palced "inside" the cell,being surrounded by other organelles.The rotation is based on the middle of the cell and` organellePos`.So  the `membranecords` dont match with actuall membrane coordonates.So rotation fails Even if its not at 0,0,because i couldve made an exception with that.

Reply:
Why not then either disconnect the visual rotation from the thrust calculation *or* also change the thrust calculation to re-calculate the orientation like for the visuals, so that way the visuals and the computed thrust would match?

Reply:
Sorry,i didnt understand,I dont think the visual rotation and thrust calculation are connected?

Reply:
They aren't currently, I was just trying to explain what could be done to address this issue *without* having to modify the organelle layout. I just looked at the code and it is hugely complicated... it needs a ton of explaining comments if it is to be kept as is. That's why I even more strongly want just the calculations done on the flagella to be changed, otherwise this causes code complications in many other places.